test/codeu/chat/client/ClientUserTest.java

```diff
@@ -0,0 +1,36 @@
+//Backend Test of ClientUser method(s)
+
+package codeu.chat.client;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+
+import codeu.chat.common.User;
+import codeu.chat.util.Logger;
+import codeu.chat.util.Uuid;
+import codeu.chat.util.store.Store;
+
+public final class ClientUserTest {
+  
+  @Test
+  public void testClientUserTest() {
+  	
+  	/*
```

Comment:
Here you need to do something along the lines of the pseudocode below:

ClientUser clientUser = new ClientUser(someController, someView);
clientUser.addUser(someName);
assertTrue(getUsers().contains(someName));

clientUser.addUser(someName);
assertThat the server error was logged
assertTrue(getUsers().contains(one(someName)));

You can find JUnit assert statements here: http://junit.org/junit4/javadoc/4.8/org/junit/Assert.html#assertTrue(boolean)

You can mock out the controller and view if need be using Mockito. This just means that the test uses dummy objects instead. Read more here: http://www.vogella.com/tutorials/Mockito/article.html